### PR TITLE
CHI-1993 Modifications to Yellow banner

### PR DESCRIPTION
**Pull request by mythilytm**
Primary reviewer:

## Description
- This PR introduces significant updates to the yellow banner's text display logic and structure with shorter content and adding icons instead of the channel type string for prefixing the identifier. The changes apply in both cases whether the helpline has the client_profiles service flag turned on. (see picture below)
- This work temporarily hardcodes the number of clients (Z) until the backend supports multiple clients for one identifier.
- 'View Records' is replaced by adding links to contacts, cases, and clients respectively
- In the `0 contacts/cases` scenario, the respective string is hidden, and singular and plural forms are supported.
- Translation strings are updated across most languages. Some need attention from the product team.
- This work affects all helplines

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added
- [ ] Feature flags added 
- [x] Strings are localized
- [x] Tested for chat contacts
- [x] Tested for call contacts

### Related Issues
Fixes # https://tech-matters.atlassian.net/browse/CHI-1993

### Verification steps
Screenshot 2024-02-01 at 11 48 25 AM
^above is modica. icon is the same as sms. Ignore the lack of comma. this has been corrected in code
Screenshot 2024-02-01 at 12 02 11 PM

- Start a contact from any channel - better to test as many as possible
- Check the yellow banner for correct content and behavior in how the contacts and cases are displayed and interact (opens the correct modal)
- Check both cases of client_profiles flag - this logic can be hard-coded by switching logic [here](https://github.com/techmatters/flex-plugins/pull/2045/files#diff-eadb9912a46353712477e051ccb5e8f3f99b4ec70b43725afdac515361e7bdaf)

### CODE REVIEW

**Comment by stephenhand** [comment]
Could you put a description on the PR, just to give reviewers some context?

**Comment by mythilytm** [comment]
Sorry, this PR is not ready yet. Yes, I will update

**Comment by mythilytm** [comment]
PR is ready of review now

**Review by GPaoloni — APPROVED** [review]
Looks and works great!! :rocket:

File: plugin-hrm-form/src/components/profile/IdentifierBanner/PreviousContactsBanner.tsx [hunk 1]
```diff
@@ -177,6 +153,10 @@ const mapDispatchToProps = (dispatch, ownProps) => {
       dispatch(newOpenModalAction({ route: 'search', subroute: 'form' }, taskId));
       dispatch(changeRoute({ route: 'search', subroute: 'contact-results', contactsPage: 0, casesPage: 0 }, taskId));
     },
+    openCaseSearchResults: () => {
+      dispatch(newOpenModalAction({ route: 'search', subroute: 'form' }, taskId));
+      dispatch(changeRoute({ route: 'search', subroute: 'case-results', contactsPage: 0, casesPage: 0 }, taskId));
```
[the remark this answers is not in the document]
Comment (mythilytm) [reply plugin-hrm-form/src/components/profile/IdentifierBanner/PreviousContactsBanner.tsx:158]: I am taking a page out of this comment above for opening contact results: 
`      // We put the form 'under' the search results in the modal stack so the back button takes them to the form without needing custom handlers
`

File: plugin-hrm-form/src/components/profile/IdentifierBanner/ProfileIdentifierBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  const profileId = identifier?.profiles?.[0]?.id;
+
+  const contactsCount = useProfileProperty(profileId, 'contactsCount') || 0;
+  const casesCount = useProfileProperty(profileId, 'casesCount') || 0;
+
+  const maskIdentifiers = !can(PermissionActions.VIEW_IDENTIFIERS);
+
+  // We immediately create a contact when a task is created, so we don't want to show the banner
+  const shouldDisplayBanner = contactsCount > 0 || casesCount > 0;
+  if (!shouldDisplayBanner) return null;
```
[the remark this answers is not in the document]
Comment (mythilytm) [reply plugin-hrm-form/src/components/profile/IdentifierBanner/ProfileIdentifierBanner.tsx:72]: Might be a good discussion point to get @stephenhand's perspective. `contactsCount` is a response from the backend and does not count the in progress contact -`ExistingContactsState` as it is only in redux.

**Pull request merged**